### DISCUSSION TITLE
How do I define a default version for the regex manager ?

### DISCUSSION THREAD

**Discussion by fclairamb** [renovatebot/renovate#23571] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I created this regex manager rule:
```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "regex": {
        "enabled": true
    },
    "regexManagers": [
        {
            "fileMatch": ["Jenkinsfile"],
            "matchStrings": [
                "@Library\\(['\"](?<depName>[^'\"@]+)@(?<currentValue>[^'\"@]+)['\"]\\)"
            ],
            "datasourceTemplate": "github-tags",
            "depNameTemplate": "__org__/{{{depName}}}"
        }
    ]
}
```

It matches patterns like this:
```groovy
@Library('jenkins-lib@v1.0.0')
```

It works _fine_.

But I'd like now to deal with cases where the tag isn't specified:
```groovy
@Library('jenkins-lib')
```

Here pinning to the latest tag or even selecting an arbitrary version defined in the config would be fine.

### What you tried so far.

I tried to create empty versions but it doesn't work (it doesn't seem to match on anything)
```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "regex": {
        "enabled": true
    },
    "regexManagers": [
        {
            "fileMatch": ["Jenkinsfile"],
            "matchStrings": [
                "@Library\\(['\"](?<depName>[^'\"@]+)@?(?<currentValue>[\\d]*)['\"]\\)"
            ],
            "datasourceTemplate": "github-tags",
            "depNameTemplate": "__org__/{{{depName}}}"
        }
    ]
}
```

**Comment by rarkins** [comment]
What are you wanting to do *after* it matches if there's no version to update?

**Reply by fclairamb** [reply]
I'd like it to pin to the latest version.

**Reply by rarkins** [reply]
I'm not sure that pinning with Regex manager can work

**Reply by fclairamb** [reply]
Well, any solution would work for me. Something like replacing `@Library('jenkins-lib')` to `@Library('jenkins-lib@v1.0.0')` would be fine. I understand it could be considered outside the scope of renovate but if could magically handle it, it would be great.

**Reply by rarkins** [reply]
With flexible capabilities like regex manager or autoreplace, sometimes people can achieve things we hadn't intended/thought of, but I'm not sure if this is one of them - I don't have any answer to you. It might only work if you decide on a static/fixed version to pin to, and put that as a static string

**Reply by fclairamb** [reply]
Yes that would be fine. Do you have any example of that ?

**Reply by rarkins** [reply]
Try currentValueTemplate
